source/core/EasyLogging.hpp

```diff
@@ -0,0 +1,307 @@
+#include <iostream>
+#include <sstream>
+#include <map>
+#include <fstream>
+
+namespace clogged{
+
+/**
+ * @brief Log levels for logging
+ * 
+ */
+    enum class LogLevel {
+        DEBUG,
+        INFO,
+        WARNING,
+        ERROR,
+        NA
+    };
+
+/**
+ * @brief Teams Names for logging
+ * 
+ */
+    enum class Team {
+        TEAM_1,
+        TEAM_2,
+        TEAM_3,
+        TEAM_4,
+        TEAM_5,
+        TEAM_6,
+        TEAM_7,
+        TEAM_8,
+        TEAM_9,
+        GENERAL,
+        NA
+    };
+
+/**
+ * @brief Colors for logging
+ * 
+ */
+    enum class Color {
+        RESET = 0,
+        BLUE = 34,
+        GREEN = 32,
+        RED = 31
+    };
+
+/**
+ * @brief Level of logging
+ * @TODO: Change this to be a flag in the CMakeLists.txt
+ */
+    const LogLevel LOGLEVEL = LogLevel::DEBUG;
+
+
+/// @brief Ensure that we only log when NDEBUG flg is not set
+#ifndef NDEBUG
+
+#define LOGLINE \
+        "File: " << __FILE__ <<  "::->::Line(" << __LINE__ << ")"
+
+#define RELATIVE_PATH(file) \
+    (std::string(file).find_last_of("/\\") != std::string::npos \
+        ? std::string(file).substr(std::string(file).find_last_of("/\\") + 1) \
+        : std::string(file))
+
+#define LOG_RELLINE \
+        "File: " << RELATIVE_PATH(__FILE__) << "::->::Line(" << __LINE__ << ")"
+
+#define LOG_FNC \
+        "Function: " << __func__ << " "
+
+/**
+ * @brief Logger class with colors and team names
+ * @author @amantham20 @chatGPT
+ * 
+ * @details I was about to name this clogged.
+ * @details This is a custom logger class that can be used to log messages to the console.
+ */
```

Comment:
I recommend indenting comments to the same level as the code they are describing the make the connection clear.

Reply:
fixed format/tidying with CLANG format